File: How to transform XML using XSLT, all parameters as strings. No file system/[VB]-How to transform XML using XSLT, all parameters as strings. No file system/VB/XmlTransformation/Core/XsltTransformer.vb
Imports System.Xml.Xsl
Imports System.Xml
Imports System.IO

Namespace XmlTransformation

	Public Class XsltTransformer
		Private ReadOnly xslTransform As XslCompiledTransform

		Public Sub New(ByVal xsl As String)
			xslTransform = New XslCompiledTransform()

			Dim xslt = XmlReader.Create(New StringReader(xsl))
			xslTransform.Load(xslt)
		End Sub

		Public Function Transform(ByVal person As Object) As String
			Dim xml = New XmlSerializerHelper().Serialize(person)

			Using writer = New StringWriter()
				Dim input = XmlReader.Create(New StringReader(xml))

				xslTransform.Transform(input, Nothing, writer)
				Return writer.ToString()
			End Using
		End Function
	End Class
End Namespace

File: How to transform XML using XSLT, all parameters as strings. No file system/[C#]-How to transform XML using XSLT, all parameters as strings. No file system/C#/XmlTransformation/Core/XsltTransformer.cs
namespace XmlTransformation
{
	using System.IO;
	using System.Xml;
	using System.Xml.Xsl;

	public class XsltTransformer
	{
		private readonly XslCompiledTransform xslTransform;

		public XsltTransformer(string xsl)
		{
			xslTransform = new XslCompiledTransform();

			var xslt = XmlReader.Create(new StringReader(xsl));
			xslTransform.Load(xslt);
		}

		public string Transform(object person)
		{
			var xml =
				new XmlSerializerHelper()
					.Serialize(person);

			using (var writer = new StringWriter())
			{
				var input = XmlReader.Create(new StringReader(xml));

				xslTransform.Transform(input, null, writer);
				return writer.ToString();
			}
		}
	}
}